src/infer_context.cc

```diff
@@ -145,36 +167,45 @@ InferContext::SendRequest(
 
     total_ongoing_requests_++;
   } else {
-    std::chrono::time_point<std::chrono::system_clock> start_time_sync,
-        end_time_sync;
-    thread_stat_->idle_timer.Start();
-    start_time_sync = std::chrono::system_clock::now();
     cb::InferResult* results = nullptr;
-    thread_stat_->status_ = infer_backend_->Infer(
-        &results, *(infer_data_.options_), infer_data_.valid_inputs_,
-        infer_data_.outputs_);
+    thread_stat_->idle_timer.Start();
+    auto start_time_sync = std::chrono::system_clock::now();
+    if (streaming_) {
+      thread_stat_->status_ = infer_backend_->StreamInfer(
+          &results, *(infer_data_.options_), infer_data_.valid_inputs_,
+          infer_data_.outputs_);
+    } else {
+      thread_stat_->status_ = infer_backend_->Infer(
+          &results, *(infer_data_.options_), infer_data_.valid_inputs_,
```

Comment:
I don't think `StreamInfer` needs to exist. Especially considering all of the streaming is encapsulated below this point.

It takes the same parameters as Infer. Infer can just be used here and within the dgrpc backend, it can implement Infer the same way it currently implements StreamInfer.

Reply:
Removed the dependency to streaming boolean.